### ISSUE TITLE
Splitting RxJS-DOM

### ISSUE THREAD

**Issue by staltz** [Reactive-Extensions/RxJS-DOM#66]
RxJS-DOM can be easily partitioned into sub-libraries, since these partitions seem to be orthogonal with regard to functionality. Partitions being

- DOM Events: `fromEvent`, `ready`, event shortcuts, pointer events, touch events.
- Ajax
- Server-Sent Events
- Web Sockets
- Web Workers
- Mutation Observers
- Geolocation
- Schedulers
- FileReader

These could be respectively split into:
- `rx.dom.events.js`
- `rx.dom.ajax.js`
- `rx.dom.sse.js`
- `rx.dom.websockets.js`
- `rx.dom.webworkers.js`
- `rx.dom.mutations.js`
- `rx.dom.geolocation.js`
- `rx.dom.schedulers.js`
- `rx.dom.filereader.js`

The advantage of splitting is we can use only a subset of RxJS-DOM while not having to carry all the load of the whole library. For instance, for frameworks which already handle DOM events, like Cycle.js (and others, this isn't the only one), the DOM Events part of RxJS-DOM is not necessary, and we might want to take only the Ajax helpers.

I can help with this issue.

**Comment by staltz** [comment]
Noticing `package.json`, I see
```
  "dependencies": {
    "rx": "*"
  },
```

Which means if my package includes both `rx` and `rx-dom`, there will be two `rx` packages: the top-level one and the one inside `rx-dom`. To fix this, we should make `rx` a peerDependency in `rx-dom`.

Approved? @mattpodwysocki

**Comment by mattpodwysocki** [comment]
@staltz sure, approved

**Comment by staltz** [comment]
@mattpodwysocki just letting you know, I'm not intending to do this issue anymore, since I found another way of achieving what I needed.

**Comment by BerkeleyTrue** [comment]
I'd be interested in splitting ajax. I've already split it into a separate script in an internal project. Would be great for it to be a proper npm module.

**Comment by mattpodwysocki** [comment]
@staltz well, I did what you asked in terms of making it smaller although you don't need it. We now have rx.dom.events.js et al, just see the modules folder for more details

**Comment by staltz** [comment]
Nice , thanks! I know people who could benefit from this
